resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/YARNConfigurableCredentialManager.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.yarn.security
+
+import java.util.ServiceLoader
+
+import scala.collection.JavaConverters._
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.security.Credentials
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.security.ConfigurableCredentialManager
+import org.apache.spark.util.Utils
+
+/**
+ * This class exists for backwards compatibility.  It loads services registered under the
+ * deprecated [[org.apache.spark.deploy.yarn.security.ServiceCredentialProvider]].
+ */
+private[yarn] class YARNConfigurableCredentialManager(
+    sparkConf: SparkConf,
+    hadoopConf: Configuration)
+    extends ConfigurableCredentialManager(
+      sparkConf,
+      hadoopConf,
+      new YARNHadoopAccessManager(hadoopConf, sparkConf)) {
+
+  val deprecatedCredentialProviders = getDeprecatedCredentialProviders
+
+  def getDeprecatedCredentialProviders:
+    Map[String, org.apache.spark.deploy.yarn.security.ServiceCredentialProvider] = {
```

Comment:
I was trying to differentiate clearly between the deprecated version and the new version, but I actually just renamed the new version to `HadoopDelegationTokenProvider` to be more clear about what it does, so I'll fix this now.